src/handlers/platformProviderResponsesHandlers.ts

```diff
@@ -0,0 +1,123 @@
+import { ajv } from '../ajv';
+import { db } from '../database';
+import { isTinyPgErrorWithQueryContext } from '../types';
+import {
+  DatabaseError,
+  InputValidationError,
+} from '../errors';
+import type {
+  Request,
+  Response,
+  NextFunction,
+} from 'express';
+import type { JSONSchemaType } from 'ajv';
+import type { PlatformProviderResponse } from '../types';
+
+interface GetPlatformProviderResponseByExternalIdParams {
+  externalId: string;
+}
+const getPlatformProviderResponsesByExternalIdParamsSchema
+: JSONSchemaType<GetPlatformProviderResponseByExternalIdParams> = {
+  type: 'object',
+  properties: {
+    externalId: {
+      type: 'string',
+    },
+  },
+  required: [
+    'externalId',
+  ],
+};
+const isGetPlatformProviderResponsesByExternalIdParams = ajv.compile(
+  getPlatformProviderResponsesByExternalIdParamsSchema,
+);
+const getPlatformProviderResponsesByExternalId = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): void => {
+  if (!isGetPlatformProviderResponsesByExternalIdParams(req.query)) {
+    next(new InputValidationError(
+      'Invalid request parameters.',
+      isGetPlatformProviderResponsesByExternalIdParams.errors ?? [],
+    ));
+    return;
+  }
+  db.sql<PlatformProviderResponse>(
+    'platformProviderResponses.selectByExternalId',
+    { externalId: req.query.externalId },
+  ).then((platformProviderResponsesQueryResult) => {
+    const { rows: platformProviderResponses } = platformProviderResponsesQueryResult;
+    res.status(200)
+      .contentType('application/json')
+      .send(platformProviderResponses);
+  }).catch((error: unknown) => {
+    if (isTinyPgErrorWithQueryContext(error)) {
+      next(new DatabaseError(
+        'Error retrieving platform provider responses.',
+        error,
+      ));
+      return;
+    }
+    next(error);
+  });
+};
+
+type PostPlatformProviderResponse = Omit<PlatformProviderResponse, 'createdAt'>;
+const postPlatformProviderResponseSchema: JSONSchemaType<PostPlatformProviderResponse> = {
+  type: 'object',
+  properties: {
+    externalId: {
+      type: 'string',
+      pattern: '^\\d{9}$',
```

Comment:
I think we need to drop this pattern, too, as part of the [previous conversation](https://github.com/PhilanthropyDataCommons/service/pull/373#discussion_r1203227850)!

```suggestion
```